hyperspy/_signals/signal2d.py

```diff
@@ -522,10 +538,768 @@ def crop_image(self, top=None, bottom=None,
+            b = box(coordinate[0], coordinate[1], window_size, z.shape[0],
+                    z.shape[1])
+            p_old = np.array([0, 0])
+            p_new = get_max(z, b)
+            while np.all(p_old != p_new):
+                p_old = p_new
+                b = box(p_old[0], p_old[1], window_size, z.shape[0], z.shape[1])
+                p_new = get_max(z, b)
+            if distance(coordinate, p_new) <= distance_cutoff:
+                peaks.append(tuple(p_new))
+        peaks = np.array([np.array(p) for p in set(peaks)])
+        return peaks
+
+
+class Stat(PeakFinder2D):
+    params = []  # Consider allowing some user-defined parameters
+    values = []
+
+    def find_peaks(self, z, **kwargs):
+        """
+        Method to locate positive peaks in an image based on statistical refinement
+        and difference with respect to mean intensity.
+        Parameters
+        ----------
+        z : ndarray
+            Array of image intensities.
+        **kwargs : None
+        Returns
+        -------
+        ndarray
+            (n_peaks, 2)
+            Array of peak coordinates.
+        Notes
+        -----
+        Implemented as described in the PhD thesis of Thomas White (2009) the
+        algorithm was developed by Gordon Ball during a summer project in
+        Cambridge.
+        This version by Ben Martineau (2016), with minor modifications to the
+        original where methods were ambiguous or unclear.
+        """
+        from scipy.ndimage.filters import generic_filter
+        from scipy.ndimage.filters import uniform_filter
+        from sklearn.cluster import DBSCAN
+
+        kwargs = self.get_default_params(**kwargs)
+
+        def normalize(image):
+            """Scales the image to intensities between 0 and 1."""
+            return image / np.max(image)
+
+        def _local_stat(image, radius, func):
+            """Calculates rolling method 'func' over a circular kernel."""
+            x, y = np.ogrid[-radius:radius + 1, -radius:radius + 1]
+            kernel = x ** 2 + y ** 2 <= radius ** 2
+            stat = generic_filter(image, func, footprint=kernel)
+            return stat
+
+        def local_mean(image, radius):
+            """Calculates rolling mean over a circular kernel."""
+            return _local_stat(image, radius, np.mean)
+
+        def local_std(image, radius):
+            """Calculates rolling standard deviation over a circular kernel."""
+            return _local_stat(image, radius, np.std)
+
+        def single_pixel_desensitize(image):
+            """Reduces single-pixel anomalies by nearest-neighbor smoothing."""
+            kernel = np.array([[0.5, 1, 0.5], [1, 1, 1], [0.5, 1, 0.5]])
+            smoothed_image = generic_filter(image, np.mean, footprint=kernel)
+            return smoothed_image
+
+        def stat_binarise(image):
+            """Peaks more than one standard deviation from the mean set to one."""
```

Comment:
I'll replace my comment from the old PR here - is there any justification for using one standard deviation and not 2, 3...? Does changing it make a difference, and so should it be user-configurable?

Building on that, is there any improvement using the more robust (to outliers in the data) [median and median absolute deviation](http://eurekastatistics.com/using-the-median-absolute-deviation-to-find-outliers)? Could this be an advanced option for users to specify using medians rather than means for e.g. noisy data?